src/openvic-simulation/misc/Modifier.cpp

```diff
@@ -99,6 +107,32 @@ ModifierValue ModifierValue::operator-(ModifierValue const& right) const {
 	return ret -= right;
 }
 
+ModifierValue& ModifierValue::operator*=(fixed_point_t const& right) {
+	for (auto value : mutable_iterator(values)) {
+		value.second *= right;
+	}
+	return *this;
+}
+
+ModifierValue ModifierValue::operator*(fixed_point_t const& right) const {
+	ModifierValue ret = *this;
+	return ret *= right;
```

Comment:
I'm struggling to understand this.
You dereference the pointer to get ret and then enter the above operator which modifies the value and returns *this.
Wouldn't the original value be changed then?

Reply:
`ModifierValue ret = *this;` creates a copy of `*this` and stores it in `ret`. We need to do this as `a * b` returns a new value rather than modifying `a` or `b`.

Reply:
Could you rename ret to copy for clarity?